### ISSUE TITLE
TypeError: Invalid options argument

### ISSUE THREAD

**Issue by ViSaturn** [jcoglan/canopy#53]
Hi, I was trying to run `canopy words.peg --lang js` on a file I made containing
```
grammar Words
  root  <-  first:"foo" second:"bar" <Extension>
```

But everytime I run it I get this error
```shell
TypeError: invalid options argument
    at optsArg (/usr/lib/node_modules/canopy/node_modules/mkdirp/lib/opts-arg.js:13:11)
    at mkdirp (/usr/lib/node_modules/canopy/node_modules/mkdirp/index.js:11:10)
    at write (/usr/lib/node_modules/canopy/bin/canopy:32:5)
    at Object.<anonymous> (/usr/lib/node_modules/canopy/bin/canopy:37:3)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:79:12)
    at node:internal/main/run_main_module:17:47
```

My javascript file looks like this
```js
var words = require('./words');

var types = {
  Extension: {
    convert: function() {
      return this.first.text + this.second.text.toUpperCase();
    }
  }
};

words.parse('foobar', {types: types}).convert()
// -> 'fooBAR'
```

It's the exact same as the example at the end of the page http://canopy.jcoglan.com/langs/javascript.html

What I tried:
- A different .peg file (same error)
- Reinstalling canopy from npm (same error)

**Comment by jcoglan** [comment]
This is a duplicate of #52, #48, #46, #43, #40, #38, #37.

**Comment by ViSaturn** [comment]
oops

**Comment by jcoglan** [comment]
This should now be fixed as part of the v0.4 release; see https://blog.jcoglan.com/2022/04/01/announcing-canopy-0-4/
